Automattic/wp-calypso
# Framework: Upgrade to eslint-plugin-wpcalypso v1.3.2

This pull request seeks to upgrade our custom ESLint rules package from 1.1.3 to 1.3.2, including a bug fix and two new rules (`jsx-gridicon-size` and `jsx-classname-namespace`). The first of these replaces our existing `bin/check-gridicon-format` script and the second enforces our CSS namespace guidelines.

[See changelog for full details](https://github.com/Automattic/eslint-plugin-wpcalypso/blob/master/CHANGELOG.md)

__Testing instructions:__

The application should not be affected by this change. Note that running ESLint on problematic files details warnings for new and revised rules.

```
./node_modules/.bin/eslint client/layout/index.jsx 
```
```
/Users/andrew/Documents/Code/wp-calypso/client/layout/index.jsx
  196:23  warning  className should follow CSS namespace guidelines (expected layout__ prefix)  wpcalypso/jsx-classname-namespace
  201:24  warning  className should follow CSS namespace guidelines (expected layout__ prefix)  wpcalypso/jsx-classname-namespace
  202:26  warning  className should follow CSS namespace guidelines (expected layout__ prefix)  wpcalypso/jsx-classname-namespace

✖ 3 problems (0 errors, 3 warnings)
```

---

```
./node_modules/.bin/eslint client/components/author-selector/index.jsx 
```
```
/Users/andrew/Documents/Code/wp-calypso/client/components/author-selector/index.jsx
   83:64   warning  Gridicon size should be one of recommended sizes: 12, 18, 24, 36, 48, 54, 72  wpcalypso/jsx-gridicon-size
   92:61   warning  '?' should be placed at the beginning of the line                             operator-linebreak
   98:11   warning  ':' should be placed at the beginning of the line                             operator-linebreak
  100:119  warning  '?' should be placed at the beginning of the line                             operator-linebreak
  101:29   warning  ':' should be placed at the beginning of the line                             operator-linebreak
  106:8    warning  className should follow CSS namespace guidelines (expected author-selector)   wpcalypso/jsx-classname-namespace
  177:5    warning  JSX props should not use .bind()                                              react/jsx-no-bind

✖ 7 problems (0 errors, 7 warnings)
```

@mtias @johnHackworth @deBhal 

Test live: https://calypso.live/?branch=update/eslint-wpcalypso-1-3-x
